.github/workflows/nix.yml

```diff
@@ -0,0 +1,29 @@
+name: nix
+on:
+  workflow_dispatch:
+  merge_group:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
```

Comment:
```suggestion
on:
  workflow_dispatch:
  merge_group:
  pull_request:
    branches:
      - main
    paths-ignore:
      - 'docs/**'
      - '.github/workflows/docs-*.yml'
  push:
    branches:
      - main
    paths-ignore:
      - 'docs/**'
      - '.github/workflows/docs-*.yml'
```

Reply:
The builds do nothing if no source files are modified.